libs/estdlib/src/gen_server.erl

```diff
@@ -314,36 +314,23 @@ cast(Pid, Request) when is_pid(Pid) ->
 %%          function can be safely ignored.
 %% @end
 %%-----------------------------------------------------------------------------
--spec reply(From :: from(), Reply :: term) -> term().
-reply({Pid, _Ref} = From, Reply) ->
-    Pid ! {'$reply', From, Reply},
+-spec reply(from(), Reply :: term) -> term().
+reply({Pid, Ref}, Reply) ->
+    Pid ! {Ref, Reply},
```

Comment:
This change is interesting, @fadushin where `$reply` comes from?
I did a quick check and I found so far:
- `:gen_server.reply({x, make_ref()}, :test1)` -> `{#Reference<...>, :test1}`
- `:gen_server.cast(x, :test1)` -> `{:"$gen_cast", :test1}`
- `:gen_server.call(x, :test1)` -> `{:"$gen_call", {#PID<...>, #Reference<...>}, :test1}`

Reply:
Now I understand where the flush code ("memory leak workaround") comes from.
https://github.com/atomvm/AtomVM/blob/c5e9e1c7fbb0e11e693907f1f49156d307e48da1/libs/estdlib/src/gen_server.erl#L343-L346

Few comments regarding the message format change:
- as mentioned elsewhere in the code of this PR, OTP's man page specifies the format of messages to allow clients to flush timeout messages. So let's stick to this specification. A quick investigation shows this specification was in `gen_server` man pages [since at least R13B03](https://github.com/erlang/otp/blob/84adefa331c4159d432d22840663c38f155cd4c1/lib/stdlib/doc/src/gen_server.xml#L211-L213)
- old format with `'$reply'` actually predates R13B03;
- in most other cases, while tests pass with OTP, we don't need to stick to internal formats used by OTP.

Now, regarding the memory leak workaround:
- normal behavior is to terminate the caller process with reason timeout, in which case the late message is not sent (it's the VM's responsibility to not leak here, and I believe it doesn't);
- if the client decides to catch the timeout reason, it is its responsibility to remove the message from its queue, or to terminate and let the VM clear it. This flush is what is mentioned in OTP gen_server(3)'s man page;
- the memory leak you experienced probably is #365: when the process terminates, the mailbox is not emptied. It's the VM's reponsibility, not estdlib's, and it is fixed in #368.

Reply:
Excellent analysis.  So we will have a memory leak until we merge SMP, is that correct?  Sounds okay to me.  (All the more reason to merge SMP ASAP)